sale_stock_available_to_promise_release_block/wizards/unblock_release.py

```diff
@@ -0,0 +1,102 @@
+    _description = "Unblock Release"
+
+    order_line_ids = fields.Many2many(
+        comodel_name="sale.order.line",
+        string="Order Lines",
+    )
+    move_ids = fields.Many2many(
+        comodel_name="stock.move",
+        string="Delivery moves",
+    )
+    option = fields.Selection(
+        selection=lambda self: self._selection_option(),
+        default="asap",
+        required=True,
+    )
+    date_deadline = fields.Datetime(
+        compute="_compute_date_deadline", store=True, readonly=False, required=True
+    )
+
+    _sql_constraints = [
+        (
+            "check_scheduled_date",
+            "CHECK (date_deadline::date >= now()::date)",
+            "You cannot reschedule deliveries in the past.",
+        ),
+    ]
+
+    def _selection_option(self):
+        options = [
+            ("free", "Free"),
+            ("asap", "As soon as possible"),
+        ]
+        if self.env.context.get("from_sale_order_id"):
+            options.append(("contextual", "Contextual"))
+        return options
+
+    @api.depends("option")
+    def _compute_date_deadline(self):
+        from_sale_order_id = self.env.context.get("from_sale_order_id")
+        order = self.env["sale.order"].browse(from_sale_order_id).exists()
+        for rec in self:
+            rec.date_deadline = False
+            if rec.option == "asap":
+                rec.date_deadline = fields.Datetime.now()
+            elif rec.option == "contextual" and order:
+                rec.date_deadline = order.commitment_date or order.expected_date
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        active_model = self.env.context.get("active_model")
+        active_ids = self.env.context.get("active_ids")
+        from_sale_order_id = self.env.context.get("from_sale_order_id")
+        from_sale_order = self.env["sale.order"].browse(from_sale_order_id).exists()
+        if active_model == "sale.order.line" and active_ids:
+            res["order_line_ids"] = [(6, 0, active_ids)]
+        if active_model == "stock.move" and active_ids:
+            res["move_ids"] = [(6, 0, active_ids)]
+        if from_sale_order:
+            res["option"] = "contextual"
+        return res
+
+    def validate(self):
+        self.ensure_one()
+        move_states = (
+            "draft",
+            "waiting",
+            "confirmed",
+            "partially_available",
+            "assigned",
+        )
+        moves = (self.order_line_ids.move_ids or self.move_ids).filtered_domain(
+            [("state", "in", move_states), ("release_blocked", "=", True)]
+        )
+        # Unset current deliveries (keep track of them to delete empty ones at the end)
+        pickings = moves.picking_id
+        moves.picking_id = False
+        # Update the scheduled date
+        date_deadline = (
+            fields.Datetime.now() if self.option == "asap" else self.date_deadline
+        )
```

Comment:
already done in `_compute_date_deadline`
```suggestion
        date_deadline = self.date_deadline
```